controller/program/tests/tests/protocol_fee/v2/withdraw_protocol_fees.rs

```diff
@@ -0,0 +1,535 @@
+    pool: PoolStateV2,
+    inf_mint_supply: u64,
+    withdraw_to_balance: u64,
+) -> AccountMap {
+    const LAMPORTS: u64 = 1_000_000_000;
+
+    let token_prog = *keys.token_program();
+    let inf_mint_pk = *keys.inf_mint();
+
+    let accs = NewWithdrawProtocolFeesV2IxAccsBuilder::start()
+        .with_pool_state(pool_state_v2_account(pool))
+        .with_beneficiary(mock_sys_acc(LAMPORTS))
+        .with_withdraw_to(mock_token_acc_with_prog(
+            raw_token_acc(inf_mint_pk, *keys.beneficiary(), withdraw_to_balance),
+            token_prog,
+        ))
+        .with_inf_mint(mock_mint_with_prog(
+            gen_inf_mint(inf_mint_supply),
+            token_prog,
+        ))
+        .with_token_program(ALL_FIXTURES.get(&TOKENKEG_PROGRAM.into()).unwrap().clone())
+        .build();
+
+    keys.0.into_iter().map(Into::into).zip(accs.0).collect()
+}
+
+fn withdraw_protocol_fees_v2_test(
+    ix: Instruction,
+    bef: &AccountMap,
+    expected_err: Option<impl Into<ProgramError>>,
+) {
+    let [pool_pk, withdraw_to_pk, inf_mint_pk] = [
+        WITHDRAW_PROTOCOL_FEES_V2_IX_ACCS_IDX_POOL_STATE,
+        WITHDRAW_PROTOCOL_FEES_V2_IX_ACCS_IDX_WITHDRAW_TO,
+        WITHDRAW_PROTOCOL_FEES_V2_IX_ACCS_IDX_INF_MINT,
+    ]
+    .map(|i| ix.accounts[i].pubkey);
+    let result = SVM.with(|svm| mollusk_exec(svm, &[ix], bef));
+
+    match expected_err {
+        None => {
+            let aft: AccountMap = result.unwrap().resulting_accounts;
+
+            let [pool_bef, pool_aft] = {
+                acc_bef_aft(&pool_pk, bef, &aft).map(|acc| {
+                    PoolStateV2Packed::of_acc_data(&acc.data)
+                        .unwrap()
+                        .into_pool_state_v2()
+                })
+            };
+
+            let [withdraw_to_bef, withdraw_to_aft] = {
+                acc_bef_aft(&withdraw_to_pk, bef, &aft)
+                    .map(|acc| RawTokenAccount::of_acc_data(&acc.data).unwrap())
+            };
+
+            let [inf_mint_bef, inf_mint_aft] = {
+                acc_bef_aft(&inf_mint_pk, bef, &aft).map(|acc| {
+                    RawMint::of_acc_data(&acc.data)
+                        .and_then(Mint::try_from_raw)
+                        .unwrap()
+                })
+            };
+
+            let inf_calc = InfCalc {
+                pool_lamports: PoolSvLamports::from_pool_state_v2(&pool_bef),
+                mint_supply: inf_mint_bef.supply(),
+            };
+            let expected_minted = inf_calc.sol_to_inf(pool_bef.protocol_fee_lamports).unwrap();
+
+            assert_token_acc_diffs(
+                withdraw_to_bef,
+                withdraw_to_aft,
+                &token_acc_bal_diff_changed(withdraw_to_bef, expected_minted as i128),
+            );
+
+            assert_eq!(
+                inf_mint_aft.supply() - inf_mint_bef.supply(),
+                expected_minted
+            );
+
+            assert_eq!(pool_aft.protocol_fee_lamports, 0);
```

Comment:
nit: to assert that we didn't accidentally clobber any other fields

```suggestion
            assert_diffs_pool_state_v2(
                &DiffsPoolStateV2 {
                    u64s: PoolStateV2U64s::default()
                        .with_protocol_fee_lamports(Diff::Changed(pool_bef.protocol_fee_lamports, 0)),
                    ..Default::default()
                },
                &pool_bef,
                &pool_aft,
            );
```